confighelper/beaconbroadcaster.py

```diff
@@ -0,0 +1,129 @@
+GROUP_RO_MODE = stat.S_IXUSR | stat.S_IRUSR | stat.S_IRGRP
+RW_MODE = stat.S_IXUSR | stat.S_IRUSR | stat.S_IWUSR
+
+BEACON_PATH = '/var/spool/'
+TMPFS_SIZE = '1M'
+
+class BeaconBroadcasterBroadcastException(Exception):
+    """ This exception is raised when a BeaconBroadcaster object fails to send a beacon."""
+
+class BeaconBroadcasterInitException(Exception):
+    """ This exception is raised when a BeaconBroadcaster object fails to initialize."""
+
+class BeaconBroadcaster(object):
+    """ Provides the broadcasting component of a filesystem-based IPC mechanism."""
+
+    def __init__(self, program_name, beacon_name, uid, gid):
+        """ Initial configuration of the beacon directory. This must be done as root.
+
+        program_name: The name of the program broadcasting the beacon.
+        beacon_name: The name of the beacon to broadcast.
+        uid: The UID of the calling program
+        gid: The GID of the calling program
+        """
+
+        self.logger = logging.getLogger(__name__)
+
+        self.logger.debug("Initializing broadcaster for beacon %s from program %s.",
+                          beacon_name, program_name)
+
+        self.beacon_path = os.path.join(BEACON_PATH, program_name, beacon_name)
+        self.program_name = program_name
+        self.beacon_name = beacon_name
+
+        tmpfs_path = os.path.join(BEACON_PATH, program_name)
+        tmpfs.mount_tmpfs(tmpfs_path, TMPFS_SIZE)
+
+        self._create_directory(self.beacon_path)
+        self._set_file_permissions(self.beacon_path, GROUP_RW_MODE, uid, gid)
+
+        self.logger.info("Broadcaster for eacon %s from program %s initialized.",
+                         beacon_name, program_name)
+
+    def send(self):
+        """ Place a new file in the beacon directory. Will raise an exception if it fails."""
+        self.logger.info(
+            "Sending beacon %s for program %s.", self.beacon_name, self.program_name)
+        now = datetime.datetime.now().isoformat()
+        random_number = os.urandom(16).encode('hex')
+
+        beacon_filename = '%s---%s' % (now, random_number)
+        final_path = os.path.join(self.beacon_path, beacon_filename)
+
+        try:
+            previous_beacons = os.listdir(self.beacon_path)
+            open(final_path, 'a').close()
+
+            for beacon_file in previous_beacons:
+                os.remove(os.path.join(self.beacon_path, beacon_file))
+
+        except Exception as exception:
+            message = 'Could not create beacon file for beacon %s, program %s. %s:%s' % (
+                self.beacon_name, self.program_name, type(exception).__name__,
+                str(exception))
+            self.logger.critical(message)
+            raise BeaconBroadcasterBroadcastException(message)
+
+    def _set_file_permissions(self, path, mode, uid, gid):
+        """ Sets permissions for a file. Will raise an exception if it fails.
```

Comment:
Two spaces behind a period in PEP-8 style block comments. (One space everywhere else, because who does that?)